pupil_src/shared_modules/square_marker_detect.py

```diff
@@ -301,24 +319,197 @@ def detect_markers_robust(gray_img,grid_size,prev_markers,min_marker_perimeter=4
     prev_img = gray_img.copy()
     return markers
 
-
-
-def bench():
-    cap = cv2.VideoCapture('/Users/mkassner/Pupil/datasets/markers/many.mov')
+def int_to_bin_list(value, width=None):
+    """Create binary repr from int
+
+    from http://stackoverflow.com/questions/22227595/convert-integer-to-binary-array-with-suitable-padding
+    """
+    m = width or 6
+    return (((value & (1 << np.arange(m)))) > 0).astype(int)
+
+def bin_list_to_int(bin_list):
+     """Create int from binary repr
+
+     from http://stackoverflow.com/questions/26060839/convert-a-binary-string-into-signed-integer-python
+     """
+     return (bin_list<<range(len(bin_list))).sum(0)
+
+class MarkerTracker(object):
+    """docstring for MarkerTracker"""
+
+    vert_slc = slice(0,8)
+    loc_conf_idx = 8
+    id_slc = slice(9,15)
+
+    def __init__(self, detect_func=detect_markers):
+        super(MarkerTracker, self).__init__()
+        self.detect_in_frame = detect_func
+        self.history = []
```

Comment:
I do not think that `state` is the best name for it. `History` is not either. Especially if we have observations in a similar structure.